api/src/wfl/module/covid.clj

```diff
@@ -157,34 +204,110 @@
 
 ;; Create and add new snapshots to the snapshot queue
 (defn ^:private update-tdr-source [source]
-  (letfn [(find-new-rows       [source now]      [])
-          (make-snapshots      [source row-ids])
-          (write-snapshots     [source snapshots])
+  (letfn [(find-new-rows [source now] [])
+          (make-snapshots [source row-ids])
+          (write-snapshots [source snapshots])
           (update-last-checked [now])]
     (let [now     (OffsetDateTime/now)
           row-ids (find-new-rows source now)]
       (when-not (empty? row-ids)
         (write-snapshots source (make-snapshots source row-ids)))
       (update-last-checked source now))))
 
-(defn ^:private load-tdr-source [{:keys [source-ptr] :as workload}]
-  {:type    tdr-source-type
-   :queue   nil
-   :updated nil})
-
-;; maybe we want to split check-request and create to avoid writing
-;; records for bad workload requests?
-(defn ^:private create-tdr-source [source-request]
-  (letfn [(check-request! [request] (throw (Exception. "Not implemented")))
-          (make-record    [request] nil)]
-    (check-request! source-request)
-    (let [foreign-key (make-record source-request)]
-      (load-tdr-source {:source-ptr foreign-key}))))
-
-(defoverload workloads/start-workload! pipeline start-covid-workload!)
-
-(defoverload create-source! tdr-source-type create-tdr-source)
-(defoverload peek-queue!    tdr-source-type peek-tdr-source-queue)
-(defoverload pop-queue!     tdr-source-type pop-tdr-source-queue)
+(def ^:private tdr-source-serialized-fields
+  {:dataset :dataset
+   :table   :dataset_table
+   :column  :table_column_name})
+
+(defn ^:private load-tdr-source [tx {:keys [source_items] :as details}]
+  (if-let [id (util/parse-int source_items)]
+    (-> (load-record-by-id! tx "TerraDataRepoSource" id)
```

Comment:
Would instead use `tdr-source-type`.

Reply:
This has given rise to some inconsistencies. Thank you for drawing my attention to it